src/api/Services/RoleOperations.cs

```diff
@@ -128,13 +128,15 @@ internal IList<StorageAccountAndContainers> GetAccessibleContainersForPrincipal(
 				if (fsr == null)
 				{
 					// Check for cached storage account properties for the storage account name
-					var val = storageAccountProperties.Value.FirstOrDefault(x => x.StorageAccountName == storageAccountName);
+					var val = (storageAccountProperties is null) ? null : storageAccountProperties.Value.FirstOrDefault(x => x.StorageAccountName == storageAccountName);
 
 					// Check for the friendly name in the cache. If it doesn't exist, get it from Azure and add it to the cache.
-					var fname = (val is null) ? String.Empty : val.FriendlyName;
+					var fname = (val is null) ? null : val.FriendlyName;
 					if (val is null)
 					{
-							fname = rgo.GetAccountResourceTagValue(storageAccountName, Configuration.StorageAccountFriendlyTagNameKey);
+							// Get the friendly name from Azure
+							// If there is no friendly name tag specified in settings configuration, this will end up displaying the storage account name instead using the property definition
+							fname = (String.IsNullOrWhiteSpace(Configuration.StorageAccountFriendlyTagNameKey)) ? null : rgo.GetAccountResourceTagValue(storageAccountName, Configuration.StorageAccountFriendlyTagNameKey);

```

Comment:
I don't think this check does a lot of good. If the called method returns null in case its `tagName` param is null (see above comment), then this can be simplified.